autoscheduler/frontend/src/types.ts

```diff
@@ -1,3 +1,48 @@
+/* eslint-disable import/prefer-default-export */
+
 /**
  * This file is where we will declare the types of each data-holding class we will be using.
  */
+export enum MeetingType {
+  LEC, LAB, RES, INS
+}
+
+export interface Meeting {
+  id: number;
+  crn: number;
+  building: string | null;
+  meetingDays: boolean[];
+  startTimeHours: number;
+  startTimeMinutes: number;
+  endTimeHours: number;
+  endTimeMinutes: number;
+  meetingType: MeetingType;
+  section: Section;
+}
+
+export interface Section {
+  id: number;
+  subject: string;
+  courseNum: number;
+  sectionNum: number;
+  minCredits: number;
+  maxCredits: number | null;
+  currentEnrollment: number;
+  instructor: number;
+}
+
+export interface Course {
+  id: string;
+  subject: string;
+  courseNum: number;
+  title: string;
+  description: string;
+  coreCurriculum: string;
```

Comment:
Will we need `coreCurriculum`? We don't have any plans for incorporating it at the moment